docs/website/docs/general-usage/credentials/configuration.md

```diff
@@ -0,0 +1,422 @@
+---
+title: Secrets and Configs
+description: Overview secrets and configs
+keywords: [credentials, secrets.toml, secrets, config, configuration, environment
+      variables]
+---
+
+# Secrets and Configs
+
+## General Usage and an Example
+
+This is what we had in mind, when designing:
+
+1. Adding configuration and secrets to sources and resources should be no-effort.
+2. You can reconfigure the pipeline for production after it is deployed. Deployed and local code should
+   be identical.
+3. You can always pass configuration values explicitly and override any default behavior (ie. naming of the configuration keys).
+
+
+The source below reads selected tabs from Google Sheets and demonstrates a few options you have to pass config and secret values:
+
+```python
+@dlt.source
+def google_sheets(
+    spreadsheet_id,
+    tab_names=dlt.config.value,
+    credentials=dlt.secrets.value,
+    only_strings=False
+):
+    sheets = build('sheets', 'v4', credentials=Services.from_json(credentials))
+    tabs = []
+    for tab_name in tab_names:
+        data = sheets.get(spreadsheet_id, tab_name).execute().values()
+        tabs.append(dlt.resource(data, name=tab_name))
+    return tabs
+```
+### Wrong approach
+```python
+# WRONG!:
+# provide all values directly - wrong but possible.
+# secret values should never be present in the code!
+data_source = google_sheets(
+    "23029402349032049",
+    ["tab1", "tab2"],
+    credentials={"private_key": ""}
+)
+```
+:::caution
+Be careful not to put your credentials directly in the code.
+:::
+
+### Right approach
+Provide config values directly and secrets via automatic [injection mechanism](#injection-mechanism).
```

Comment:
For clarity this needs to be rephrased: "provide config values directly" is to pass them as arguments. Here we need to tell that the proper way is not to hardcode your credentials, but store them in the config files. And then explain what happens then (the values are injected).